Fix linker error on linux

`ADDITIONAL_LIBS += $(shell pkg-config fftw3 --libs)`  in Magpie's `fftw3.mk` is only applied when linking the executable. When compiling Marmot with Magpie on liinux this flag somehow does not make it through to Marmot resulting in a linker error:

```
Building in /home/schwd/Programs/marmot...
Checking if header needs updating: /home/schwd/Programs/moose/framework/include/base/MooseRevision.h...
Linking Executable /home/schwd/Programs/marmot/marmot-opt...
/usr/bin/ld: /home/schwd/Programs/magpie/lib/libmagpie-opt.so: undefined reference to `fftw_plan_r2r'
/usr/bin/ld: /home/schwd/Programs/magpie/lib/libmagpie-opt.so: undefined reference to `fftw_execute'
/usr/bin/ld: /home/schwd/Programs/magpie/lib/libmagpie-opt.so: undefined reference to `fftw_destroy_plan'
collect2: error: ld returned 1 exit status
make: *** [/home/schwd/Programs/moose/framework/app.mk:393: /home/schwd/Programs/marmot/marmot-opt] Error 1
```

The workaround seems to be to add the fftw library to `LDFLAGS` instead.

Tagging @permcody

--- a/contrib/fftw3.mk
+++ b/contrib/fftw3.mk
@@ -6,5 +6,7 @@
 ifeq ($(shell pkg-config fftw3 && echo go),go)
 ADDITIONAL_INCLUDES += $(shell pkg-config fftw3 --cflags)
 ADDITIONAL_CPPFLAGS += -DFFTW3_ENABLED
-ADDITIONAL_LIBS += $(shell pkg-config fftw3 --libs)
+FFTW3LIBS := $(shell pkg-config fftw3 --libs)
+ADDITIONAL_LIBS += $(FFTW3LIBS)
+LDFLAGS += $(FFTW3LIBS)
 endif

